pkg/rtc/transport.go

```diff
@@ -400,6 +431,19 @@ func (t *PCTransport) createAndSendOffer(options *webrtc.OfferOptions) error {
 	t.negotiationState = negotiationStateClient
 	t.restartAfterGathering = false
 
+	negotiateVersion := t.negotiateCounter.Inc()
+	if t.signalStateCheckTimer != nil {
+		t.signalStateCheckTimer.Stop()
+	}
+	t.signalStateCheckTimer = time.AfterFunc(negotiationFailedTimout, func() {
+		if t.negotiateCounter.Load() == negotiateVersion && t.negotiationState != negotiationStateNone {
+			t.params.Logger.Infow("negotiation timeout")
+			if t.onNegotiationFailed != nil {
+				t.onNegotiationFailed()
+			}
+		}
```

Comment:
Should timer be set to `nil` here? Maybe `stop` also (guess stop may not be necessary as it has already fired)?

Should this grab the lock as it is reading `negotiationState`?

Reply:
will add a lock. I think don't need to stop a fired timer